### PR TITLE
[Core] moving check to model part and removing processinfo from arguments

### PR DESCRIPTION
**Pull request by ddiezrod**
**📝 Description**
As shown in https://github.com/KratosMultiphysics/Kratos/issues/7713 , right now there is a bit of confusion about who should Check the elements/conditions etc. Right this check is actually being called twice, once in the solving strategy and once again in the scheme. The solution proposed here is to call the model part check only from the strategy and add there the checks for the entities. I think this should not break anything unless there is a strategy somewhere that is not calling the base class (I did not find any)

**🆕 Changelog**
- Removed elements/conditions/constraints checks from solving_strategy and scheme base classes.
- Update model part Check so it does exactly what the scheme used to do.
- Call model part check from scheme.

### CODE REVIEW

File: kratos/sources/model_part.cpp [hunk 1]
```diff
@@ -2203,27 +2203,36 @@ void ModelPart::SetBufferSizeSubModelParts(ModelPart::IndexType NewBufferSize)
...
-        const auto& r_constraint = *constraint_iterator;
-        err = r_constraint.Check(rCurrentProcessInfo);
-    }
-    return err;
+    //TODO: This is required for the exception handling. It can be removed once we move to the C++ parallelism
+    #ifdef KRATOS_SMP_CXX11
+        int num_threads = ParallelUtilities::GetNumThreads();
+    #else
+        int num_threads = 1;
+    #endif
```
Comment (philbucher) [inline kratos/sources/model_part.cpp]: can be removed since #9335

**Review by philbucher — DISMISSED** [review]
makes perfect sense to me :+1:

File: kratos/sources/model_part.cpp (hunk 1 above)
Comment (ddiezrod) [reply kratos/sources/model_part.cpp]: Oh yes, I forgot about that ;)

File: kratos/sources/model_part.cpp [hunk 2]
```diff
@@ -2203,27 +2203,31 @@ void ModelPart::SetBufferSizeSubModelParts(ModelPart::IndexType NewBufferSize)
...
+
+    // Checks for all of the conditions
+    BlockPartition<const ModelPart::ConditionsContainerType>(this->Conditions(), num_threads).for_each([&r_current_process_info](const Condition& rCondition){
+        rCondition.Check(r_current_process_info);
+    });
+
+    // Checks for all of the constraints
+    BlockPartition<const ModelPart::MasterSlaveConstraintContainerType>(this->MasterSlaveConstraints(), num_threads).for_each([&r_current_process_info](const MasterSlaveConstraint& rConstraint){
+        rConstraint.Check(r_current_process_info);
+    });
```
Comment (philbucher) [inline kratos/sources/model_part.cpp]:
```suggestion
    block_for_each(this->MasterSlaveConstraints(), [&r_current_process_info](const MasterSlaveConstraint& rConstraint){
        rConstraint.Check(r_current_process_info);
    });
```

same for the others

Comment (ddiezrod) [reply kratos/sources/model_part.cpp]: done 👍

**Review by philbucher — APPROVED** [review]
perfect :+1:

**Comment by ddiezrod** [comment]
merging ;)

**Pull request merged**